docs/diagrams/TasksOnSequenceDiagram.puml

```diff
@@ -0,0 +1,84 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant ":TasksOnCommandParser" as TasksOnCommandParser LOGIC_COLOR
+participant "<<class>>\nDateTime" as DateTimeClass LOGIC_COLOR
+participant ":DateTime" as DateTime LOGIC_COLOR
+participant "t:TasksOnCommand" as TasksOnCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("tasksOn 22-10-2022")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("tasksOn 22-10-2022")
+activate AddressBookParser
+
+create TasksOnCommandParser
+AddressBookParser -> TasksOnCommandParser
+activate TasksOnCommandParser
+
+TasksOnCommandParser -> DateTimeClass : ofDate("22-10-2022")
+activate DateTimeClass
+
+create DateTime
+DateTimeClass -> DateTime : new DateTime("22-10-2022")
+activate DateTime
+
+DateTime -> DateTimeClass : dt
+deactivate DateTime
+
+DateTimeClass -> TasksOnCommandParser : dt
+deactivate DateTimeClass
+
+TasksOnCommandParser -> TasksOnCommand : dt
+activate TasksOnCommand
+
+TasksOnCommand -> TasksOnCommandParser : t
+deactivate TasksOnCommand
+
+TasksOnCommandParser -> AddressBookParser : t
+deactivate TasksOnCommandParser
+TasksOnCommandParser -[hidden]-> AddressBookParser
+destroy TasksOnCommandParser
+
+LogicManager -> TasksOnCommand : execute()
+activate TasksOnCommand
```

Comment:
I think after `TaskOnCommandParser` is destroyed, control should be handed back to `UninurseBookParser` first before going back to `LogicManager`, hence the activation bar should not extend all the way.

Reply:
done